app/main/routes.py

```diff
@@ -708,3 +711,179 @@ def terms_of_use():
 @bp.app_errorhandler(HTTPException)
 def http_exception(error):
     return render_template(f"{error.code}.html"), error.code
+
+
+@bp.route("/generate_manifest/<uuid:record_id>")
+@access_token_sign_in_required
+def generate_manifest(record_id: uuid.UUID):
```

Comment:
if we do need to keep the endpoint, would it be better to define the url as something like /record/id/manifest, like we have record/id/download etc

Reply:
This URL has been renamed to record/<id>/manifest. It is protected and does require authentication to be accessed.

Reply:
for visibility: here https://iiif.io/guides/using_iiif_resources/ is an explanation of the need for a manifest and uri.

A manifest is the crucial resource for the iiif standard.

We are choosing to use universal viewer to render iiif resources in our webapp but by exposing this authenticated manifest url, we are giving our users the ability to go and render the manifest in another viewer / look at the raw details available in the manifest. Initially I was concerned that there could be some data here we would not want to expose but I believe that it is ok; the viewer is just a convenience; all the data is in the webapage already.

Would be worth sitting down and verifying this is the case and letting the team know that this endpoint will be available. Whether we want to advertise this endpoint is an interesting question.

Reply:
Discussed with AYR, ok to leave this protected route in.